dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/mail/MailService.java

```diff
@@ -0,0 +1,148 @@
+        mailSender = MailerBuilder
+                .usingSession(session)
+                .withDebugLogging(true)
+                .buildMailer();
+    }
+
+    // -------------------- LOGIC --------------------
+
+    /**
+     * Gathers template messages for given notification and sends email.
+     *
+     * @return true if email was sent or false if some error occurred and email could not be sent.
+     */
+    public Boolean sendNotificationEmail(EmailNotificationDto notification, Optional<AuthenticatedUser> requestor) {
+
+        String userEmail = notification.getUserEmail();
+        String systemEmail = settingsService.getValueForKey(Key.SystemEmail);
+
+        Tuple2<String, String> messageAndSubject = mailMessageCreator.getMessageAndSubject(notification, requestor, systemEmail);
+
+        if (messageAndSubject._1().isEmpty() && messageAndSubject._2().isEmpty()) {
+            return false;
+        }
+
+        return sendMail(userEmail, messageAndSubject._2(), messageAndSubject._1());
+    }
+
+    /**
+     * Sends email(s).
+     *
+     * @param recipientsEmails - comma separated emails.
+     */
+    public boolean sendMail(String recipientsEmails, String subject, String messageText) {
+
+        String message = mailMessageCreator.createMailFooterMessage(messageText, dataverseService.findRootDataverse().getName(), getSystemAddress());
+
+        Email email = EmailBuilder.startingBlank()
+                .from(getSystemAddress())
+                .withRecipients(mailMessageCreator.createRecipients(recipientsEmails, StringUtils.EMPTY))
+                .withSubject(subject)
+                .appendText(message)
+                .buildEmail();
+
+        return Try.run(() -> mailSender.sendMail(email))
+                .map(emailSent -> true)
+                .onFailure(Throwable::printStackTrace)
+                .getOrElse(false);
+    }
+
+    /**
+     * Sends email(s) with replay email.
+     *
+     * @param recipientsEmails - comma separated emails.
+     */
+    public boolean sendMail(String replyEmail, String recipientsEmails, String subject, String messageText) {
+
+        Email email = EmailBuilder.startingBlank()
+                .from(getSystemAddress())
+                .withRecipients(mailMessageCreator.createRecipients(recipientsEmails, StringUtils.EMPTY))
+                .withSubject(subject)
+                .withReplyTo(replyEmail)
+                .appendText(messageText)
+                .buildEmail();
+
+        return Try.run(() -> mailSender.sendMail(email))
+                .map(emailSent -> true)
+                .onFailure(Throwable::printStackTrace)
+                .getOrElse(false);
+    }
+
+    // -------------------- PRIVATE --------------------
+
+    private InternetAddress getSystemAddress() {
+        String systemEmail = settingsService.getValueForKey(Key.SystemEmail);
+
+        return Try.of(() -> new InternetAddress(systemEmail, StringUtils.EMPTY, StandardCharsets.UTF_8.name()))
```

Comment:
I don't think passing `StringUtils.EMPTY` is what we want here. It should be `null` I think.

Reply:
Why would I pass null, for me it is like sabotage since if you have method arguments you don't expect nulls just values. If I pass null to external library can you be sure that it won't blow up inside? If I pass value that it expect's I know it won't.

Reply:
I understand but the thing is, when we dig into the constructor we get to setter:
`
    public void setPersonal(String name, String charset) throws UnsupportedEncodingException {
        this.personal = name;
        if (name != null) {
            this.encodedPersonal = MimeUtility.encodeWord(name, charset, (String)null);
        } else {
            this.encodedPersonal = null;
        }

    }
`
and comparision: `StringUtils.EMPTY != null` returns `true` and i don't think that's what we expect here. We already had problems with it but in our code where we could change 'name != null' to 'StringUtils.NotEmpty(name)` but here it's not possible.